src/context/basket.js

```diff
@@ -1,21 +1,35 @@
 import React, { createContext, useState } from 'react'
+import _ from 'lodash'
+import ItemList from '../pages/home/items-list'
+import Item from '../pages/item'
+
 
 
 
 export const BasketContext = createContext({
-    basket: '',
+    basket: [],
     updateBasket: () => null
 })
 
+
+  
+
 const BasketProvider = ({children}) => {
-    const [basket, setBasket] = useState ('')
+    const [basket, setBasket] = useState ([])
+
+ const incrementer = _.find(basket, basket.id )
+    const updateBasket = (ItemList) => {
+           
+
+        {incrementer === undefined ? setBasket((prevState) => [...prevState, {id: ItemList.id, quantity: 1}]) : setBasket(() => [ {id: incrementer.id, quantity: incrementer.quantity +1}])}
```

Comment:
Move this  `const incrementer = _.find(basket, basket.id )` bolck of code inside `updateBasket`

and replace `basket.id ` with `ItemList.id`